taxonomy: mushrooms etc

ArturLange opened the pull request:
### What
- Standarize mushrooms taxonomy a little
- Improve polish ingredients processing

aleene commented on taxonomies/ingredients.txt:
```diff
@@ -66878,8 +66969,15 @@ de:pencillium roqueforti (Edelschimmel)
+fr:penicillium glaucum
+la:penicillium glaucum
+pl:penicillium glaucum
+
```
Better replace this with one xx: line under the en: line, then the other languages lines are no longer necessary.

aleene commented on taxonomies/ingredients.txt:
```diff
@@ -66878,8 +66969,15 @@ de:pencillium roqueforti (Edelschimmel)
 es:Penicillium roqueforti
 fr:penicillium roqueforti
 la:pencillium roqueforti
+pl:pencillium roqueforti
```
Also replace all language variants with one xx: line under the en: line

aleene commented on taxonomies/ingredients.txt:
```diff
@@ -1220,6 +1220,7 @@ fi:Lactobacillus bulgaricus, L. bulgaricus
 it:Lactobacillus bulgaricus, L. bulgaricus
 fr:L bulgaricus, bulgaricus
 hr:lactobacillus bulgaricus
+pl:lactobacillus bulgaricus
```
Can do some cleanup with xx:

aleene reviewed (changes requested):
Thanks for your changes. You might update some entries and replace them with an xx: entry, the xx: works as a default when a language line is missing. This keeps some ingredients a bit smaller.

stephanegigandet reviewed (approved):
Thank you!

benbenben2 commented on tests/unit/ingredients_parsing.t:
```diff
@@ -568,6 +568,9 @@ my @lists = (
 	],
 
+	# Polish oils
+	["pl", "oleje roślinne (słonecznikowy, rzepakowy)", "olej roślinny słonecznikowy, olej roślinny rzepakowy"],
```
```suggestion
	["pl", "oleje roślinne (słonecznikowy, rzepakowy)", "oleje roślinne (słonecznikowy, rzepakowy)"],
```

benbenben2 commented on tests/unit/ingredients_processing.t:
```diff
@@ -1631,7 +1631,40 @@ my @tests = (
+		},
+		[
+			{
+				'id' => 'en:dried garlic',
```
```suggestion
				'id' => 'en:dried-garlic',
```

benbenben2 commented on tests/unit/ingredients_processing.t:
```diff
@@ -1631,7 +1631,40 @@ my @tests = (
+				'text' => 'czosnek suszony'
+			},
+			{
+				'id' => 'en:dried garlic',
```
```suggestion
				'id' => 'en:dried-garlic',
```

benbenben2 commented:
Hi @ArturLange! Great job!


Looking at the logs, you have 3 failed tests:
```
2023-05-15T21:06:54.1160257Z #   Failed test at tests/unit/ingredients_parsing.t line 590.
2023-05-15T21:06:54.1161041Z Wide character in print at /opt/perl/local/lib/perl5/Test2/Formatter/TAP.pm line 125.
2023-05-15T21:06:54.1161833Z #          got: 'oleje roślinne (słonecznikowy, rzepakowy)'
2023-05-15T21:06:54.1162307Z #     expected: 'olej roślinny słonecznikowy, olej roślinny rzepakowy'
```
If you want to parse them, it seems to be in Ingredients.pm, "# Vegetal oil (palm, sunflower and olive)". But it seems not be as simple as that in Polish.

```
023-05-15T21:07:06.1133742Z #   Failed test at tests/unit/ingredients_processing.t line 1711.
2023-05-15T21:07:06.1134121Z #     Structures begin differing at:
2023-05-15T21:07:06.1134694Z #          $got->[0]{id} = 'en:dried-garlic'
2023-05-15T21:07:06.1135129Z #     $expected->[0]{id} = 'en:dried garlic'
2023-05-15T21:07:06.1135393Z [
2023-05-15T21:07:06.1135580Z   {
2023-05-15T21:07:06.1135882Z     'id' => 'en:dried-garlic',
2023-05-15T21:07:06.1136206Z     'text' => 'czosnek suszony'
2023-05-15T21:07:06.1136432Z   },
2023-05-15T21:07:06.1136646Z   {
2023-05-15T21:07:06.1136930Z     'id' => 'en:dried-garlic',
2023-05-15T21:07:06.1137232Z     'text' => 'suszony czosnek'
2023-05-15T21:07:06.1137796Z   }
2023-05-15T21:07:06.1138180Z ]
```



```
2023-05-15T21:07:32.3591462Z #   Failed test at /opt/product-opener/lib/ProductOpener/Test.pm line 329.
2023-05-15T21:07:32.3591806Z # 
2023-05-15T21:07:32.3606064Z #     Structures begin differing at:
2023-05-15T21:07:32.3606558Z #          $got->{ingredients_tags}[2] = 'en:pleurotus'
2023-05-15T21:07:32.3607006Z #     $expected->{ingredients_tags}[2] = 'fr:chapelure-de-mais'
```
That one seems to be from tests/unit/expected_test_results/nutriscore/mushrooms.json. Try to run "make update_tests_results". Or copy paste the file from the actual main branch.

alexgarel commented:
@ArturLange thanks for the contribution. If it's too hard for you to fix tests, just tell it, we might try to find the time to do it.

alexgarel commented:
@benbenben2 do you want to try to finish this PR ?

ArturLange commented:
Thanks @alexgarel, it took me some time to get to know perl a little and set up the environment, but it really helped :+1: 

I fixed the tests and added preparsing to Polish language, @aleene could you review again? :slightly_smiling_face:

alexgarel commented:
Big up @ArturLange for the though work :tada:
